pages/docs/reference/compiler-plugins.md

```diff
@@ -275,3 +275,98 @@ As with all-open, add the plugin JAR file to the compiler plugin classpath and s
 -P plugin:org.jetbrains.kotlin.noarg:annotation=com.my.Annotation
 -P plugin:org.jetbrains.kotlin.noarg:preset=jpa
 ```
+
+
+## Sam-with-receiver compiler plugin
+
+The *sam-with-receiver* compiler plugin makes the first parameter of the annotated Java "single abstract method" (SAM) interface method a receiver in Kotlin. This conversion only works when the SAM interface is passed as a Kotlin lambda, both for SAM adapters and SAM constructors (see the [documentation](https://kotlinlang.org/docs/reference/java-interop.html#sam-conversions) for more details).
+
+Here is an example:
+
+```java
+public @interface SamWithReceiver {}
+
+@SamWithReceiver
+public interface TaskRunner {
+    void run(Task task);
+}
+```
+
+```kotlin
+fun test(context: TaskContext) {
+    val handler = TaskHandler { 
+        // Here 'this' is an instance of 'Task'
+        
+        println("$name is started")
+        context.executeTask(this)
+        println("$name is finished")
+    }
+}
+```
+
+### Using in Gradle
+
+The usage is the same to all-open and no-arg, except the fact that sam-with-receiver does not have any built-in presets, and you need to specify your own list of special-treated annotations.
+ 
+```groovy
+buildscript {
+    dependencies {
+        classpath "org.jetbrains.kotlin:kotlin-sam-with-receiver:$kotlin_version"
+    }
+}
+
+apply plugin: "kotlin-sam-with-receiver"
+```
+
+Or using the Gradle plugins DSL:
+
+```groovy
+plugins {
+  id "org.jetbrains.kotlin.plugin.sam.with.receiver" version "{{ site.data.releases.latest.version }}"
```

Comment:
Note that we haven't started publishing this plugin to the plugins portal yet.

Reply:
Oh, I didn't know that it has to be explicitly published. I'll remove that part then, thanks.